app.rb

```diff
@@ -44,9 +45,28 @@ class MakersBnB < Sinatra::Base
   end
 
   post '/spaces' do
-    @space = Space.create(name: params[:name], description: params[:description], price_per_night: params[:'price-per-night'])
+    Space.create(name: params[:name], user_id: session[:user].id, description: params[:description], price_per_night: params[:'price-per-night'])
     redirect '/spaces'
   end
 
+  get '/spaces/:id' do
+    p params[:id]
+    session[:space] = Space.find(params[:id])
+    @space = session[:space]
+    erb :'spaces/profile'
+  end
+
+  post '/requests' do
+    Request.create({ user_id: session[:user].id, date_requested: params['requested-date'], space_id: session[:space].id })
```

Comment:
if we add <code>has_many :requests</code> to the user.rb model we can do 
<code>user = User.find(id: session[:id])</code>
<code>user.requests.create({ date_requested: params['requested-date'], space_id: session[:space].id })</code>